### PR TITLE
Add patches for nginx 1.25.0 and corresponding nginx-tests patch

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Passes ssl tests when nginx is configured with
  `./auto/configure --with-wolfssl=/usr/local --with-http_ssl_module --with-stream --with-stream_ssl_module --with-stream_ssl_preread_module --with-http_v2_module --with-mail --with-mail_ssl_module`
And the tests that are confirmed to pass (with the patch applied):
  `TEST_NGINX_BINARY=../nginx/objs/nginx prove *ssl*`

### CODE REVIEW

**Comment by julek-wolfssl** [comment]
Depends on https://github.com/wolfSSL/wolfssl/pull/6515

**Review by kareem-wolfssl — CHANGES_REQUESTED** [review]

File: nginx-1.25.0-wolfssl.patch [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+         ngx_feature_path=
+         ngx_feature_libs="-lssl -lcrypto $NGX_LIBDL $NGX_LIBPTHREAD"
+         ngx_feature_test="SSL_CTX_set_options(NULL, 0)"
++
++        if [ $WOLFSSL != NONE ]; then
++            ngx_feature="wolfSSL library in $WOLFSSL"
++            ngx_feature_path="$WOLFSSL/include/wolfssl $WOLFSSL/include"
++            ngx_feature_incs="#ifdef HAVE_CONFIG_H
++                #include <config.h>
++            #endif
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply nginx-1.25.0-wolfssl.patch]: Confirmed not needed and removed.

File: nginx-1.25.0-wolfssl.patch [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
++#endif
+ 
+ #ifdef SSL_OP_NO_COMPRESSION
+     SSL_CTX_set_options(ssl->ctx, SSL_OP_NO_COMPRESSION);
+@@ -1455,7 +1458,8 @@ ngx_ssl_ecdh_curve(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *name)
+      * maximum interoperability.
+      */
+ 
+-#if (defined SSL_CTX_set1_curves_list || defined SSL_CTRL_SET_CURVES_LIST)
++#if (defined SSL_CTX_set1_curves_list || defined SSL_CTRL_SET_CURVES_LIST) || \
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply nginx-1.25.0-wolfssl.patch]: Confirmed not needed and removed.

**Review by kareem-wolfssl — CHANGES_REQUESTED** [review]
Code looks good.
Is it possible to fix the PR test?  It looks like it's missing a dependency to build nginx:
```
./auto/configure: error: the HTTP rewrite module requires the PCRE library.
You can either disable the module by using --without-http_rewrite_module
option, or install the PCRE library into the system, or build the PCRE library
statically from the source with nginx by using --with-pcre=<path> option.
```

**Comment by julek-wolfssl** [comment]
@kareem-wolfssl  Since I added testing on the wolfSSL side, can we ignore this test failure? I'll figure out how the tests here work and how they can be fixed.

**Review by kareem-wolfssl — APPROVED** [review]
Good point.
We may consider just getting rid of this Jenkins test now that we have Github actions continuously testing nginx.

**Pull request merged**